micrometer-core/src/main/java/io/micrometer/core/instrument/binder/hystrix/MicrometerMetricsPublisherThreadPool.java

```diff
@@ -0,0 +1,142 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.hystrix;
+
+import com.netflix.hystrix.HystrixThreadPoolKey;
+import com.netflix.hystrix.HystrixThreadPoolMetrics;
+import com.netflix.hystrix.HystrixThreadPoolProperties;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisherThreadPool;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+/**
+ * Micrometer publisher for Hystrix thread pool metrics.
+ */
+@NonNullApi
+@NonNullFields
+public class MicrometerMetricsPublisherThreadPool implements HystrixMetricsPublisherThreadPool {
+  private static final String NAME_HYSTRIX_THREADPOOL = "hystrix.threadpool";
+
+  private final MeterRegistry meterRegistry;
+  private final HystrixThreadPoolKey threadPoolKey;
+  private final HystrixThreadPoolMetrics metrics;
+  private final HystrixThreadPoolProperties properties;
+  private final HystrixMetricsPublisherThreadPool metricsPublisherForThreadPool;
+  private final Tags tags;
+
+  public MicrometerMetricsPublisherThreadPool(
+      final MeterRegistry meterRegistry,
+      final HystrixThreadPoolKey threadPoolKey,
+      final HystrixThreadPoolMetrics metrics,
+      final HystrixThreadPoolProperties properties,
+      final HystrixMetricsPublisherThreadPool metricsPublisherForThreadPool) {
+    this.meterRegistry = meterRegistry;
+    this.threadPoolKey = threadPoolKey;
+    this.metrics = metrics;
+    this.properties = properties;
+    this.metricsPublisherForThreadPool = metricsPublisherForThreadPool;
+
+    this.tags = Tags.of("key", threadPoolKey.name());
+  }
+
+  @Override
+  public void initialize() {
+    metricsPublisherForThreadPool.initialize();
+
+    Gauge.builder(metricName("threads.active.current.count"), metrics::getCurrentActiveCount)
+        .description("The approximate number of threads that are actively executing tasks.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.active.rolling.max"), metrics::getRollingMaxActiveThreads)
+        .description("Rolling max number of active threads during rolling statistical window.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.executed.cumulative.count"), metrics::getCumulativeCountThreadsExecuted)
```

Comment:
How do the three metrics of `threads.executed.cumulative.count` and `threads.rejected.cumulative.count` relate? If Bother of them added together equals the total, then we should combine them under a name liek `threads.cumulative.count` and make `executed`/`rejected` tags.

Also likely should be a `FunctionCounter`

Reply:
Done.